app/grandchallenge/reader_studies/views.py

```diff
@@ -1093,6 +1091,28 @@ def filter_queryset(self, queryset):
                 pks = queryset.values_list("pk", flat=True)
                 queryset = [x for x in self.randomized_qs if x.pk in pks]
 
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def get_object(self):
+        obj = super().get_object()
+        # retrieve the full queryset and save its shuffled version to later
+        # determine the shuffled index for this object
+        if obj.reader_study and obj.reader_study.shuffle_hanging_list:
```

Comment:
You don't need to check for `obj.reader_study` here, as that's a non nullable field on the model, so if it's not there something is very wrong and we probably want an exception.